nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/CommentManagerImpl.java

```diff
@@ -95,45 +95,47 @@ public CommentManagerImpl(CommentServiceConfig config) {
 
     @Override
     public List<DocumentModel> getComments(CoreSession session, DocumentModel docModel)
-            throws CommentSecurityException {
-        Map<String, Object> ctxMap = Collections.<String, Object> singletonMap(ResourceAdapter.CORE_SESSION_CONTEXT_KEY,
-                session);
-        RelationManager relationManager = Framework.getService(RelationManager.class);
-        Graph graph = relationManager.getGraph(config.graphName, session);
-        Resource docResource = relationManager.getResource(config.documentNamespace, docModel, ctxMap);
-        if (docResource == null) {
-            throw new NuxeoException("Could not adapt document model to relation resource ; "
-                    + "check the service relation adapters configuration");
-        }
+            throws CommentSecurityException { // throw declaration is kept for historical reasons
+        return CoreInstance.doPrivileged(docModel.getRepositoryName(), userSession -> {
```

Comment:
It's not a `userSession` but a system one. Note: it's better to give the `CoreSession` to `doPrivileged` method as it we keep track of originating user.

Reply:
Now that the `TestCommentManager` is merged in `TestCommentManagerImpl` it is even necessary (to give the `CoreSession`) or `TestCommentManagerImpl#testCreateComment` will fail